### PR TITLE
Fixed bug where logger could not create directory

### LINKED ISSUE #29: Cannot create logs directory
If there is not logs directory (and the logging flag is not passed), the program won't create it and crash

### PR DESCRIPTION
**Pull request by yavidor**
solves: https://github.com/royiro10/cogo/issues/29

Also changed the signature of CreateLogger to CreateLogger(logPath string, logFile string)

### CODE REVIEW

**Comment by yavidor** [comment]
Also added cogo binary to .gitignore

**Comment by royiro10** [comment]
please tag the issue you are solving

File: common/logger.go [hunk 1]
```diff
@@ -15,8 +16,19 @@ type Logger struct {
 	LogFile *os.File
 }
 
-func CreateLogger(logFile string) *Logger {
-	f, err := os.OpenFile(logFile, os.O_CREATE|os.O_WRONLY|os.O_APPEND, 0644)
+func CreateLogger(logPath string, logFile string) *Logger {
+	if _, err := os.Stat(logPath); os.IsNotExist(err) {
+		err := os.MkdirAll(logPath, 0755)
+		if err != nil {
+			// TODO: How to handle without a logger?
```
Comment (royiro10) [inline common/logger.go]: I think we panic.
at this point if we expected creating a logger and can't do so we would like notify the user things does not go as expected.


maybe later we can return an EmptyLogger and some how notify user an err occurred (TODO)

File: common/logger.go [hunk 2]
```diff
@@ -15,8 +16,19 @@ type Logger struct {
 	LogFile *os.File
 }
 
-func CreateLogger(logFile string) *Logger {
-	f, err := os.OpenFile(logFile, os.O_CREATE|os.O_WRONLY|os.O_APPEND, 0644)
+func CreateLogger(logPath string, logFile string) *Logger {
+	if _, err := os.Stat(logPath); os.IsNotExist(err) {
+		err := os.MkdirAll(logPath, 0755)
```
Comment (royiro10) [inline common/logger.go]:
```suggestion
		err := os.MkdirAll(logPath, 0644)
```

I think going for 644 permission is better at this point for logs. not allowing to execute log file can be great, and preventing others from changing the logs file

File: common/logger.go [hunk 3]
```diff
@@ -32,7 +44,7 @@ func CreateLogger(logFile string) *Logger {
 	writer := io.MultiWriter(os.Stdout, f)
 	handler := slog.NewTextHandler(writer, &slog.HandlerOptions{
 		AddSource: true,
-		Level:     slog.LevelDebug,
+		Level:     slog.LevelInfo,
```
Comment (royiro10) [inline common/logger.go]: maybe also getting this in the constructor, and using Info as default if no level provided?

File: common/logger.go (hunk 2 above)
Comment (yavidor) [reply common/logger.go]: Ok, just went with the default without really thinking about it

File: common/logger.go (hunk 1 above)
Comment (yavidor) [reply common/logger.go]: Fixed (probably)

File: common/logger.go (hunk 3 above)
Comment (yavidor) [reply common/logger.go]: A slight problem, there isn't a real optional parameter in go. Will look into it

Comment (yavidor) [reply common/logger.go]: Probably with ...args 
```go
func CreateLogger(logPath string, logFile string, levels ...slog.Level) *Logger {
var level slog.Level
if len(levels) > 0 {
level = levels[0]
} else {
level = slog.Levelnfo
}
```

Other than that I don't really have an idea

Comment (royiro10) [reply common/logger.go]: I think I would take a more go-like approach and go with and Option struct 

```golang

type LoggerOptions {
  LogPath string
  LogFile string
  Level   slog.Level (or int or something)
}

func CreateLogger(options *LoggerOptions)...
```

then check the struct and fill default values if needed

**Comment by yavidor** [comment]
Fixes #29

File: common/logger.go [hunk 4]
```diff
@@ -15,8 +16,24 @@ type Logger struct {
 	LogFile *os.File
 }
 
-func CreateLogger(logFile string) *Logger {
-	f, err := os.OpenFile(logFile, os.O_CREATE|os.O_WRONLY|os.O_APPEND, 0644)
+func CreateLogger(logPath string, logFile string, levels ...slog.Level) *Logger {
```
Comment (royiro10) [inline common/logger.go]: I think it will be better if it will come from an option struct

File: common/logger.go [hunk 5]
```diff
@@ -15,8 +16,24 @@ type Logger struct {
...
+	var level slog.Level
+	if len(levels) > 0 {
+		level = levels[0]
+	} else {
+		level = slog.LevelInfo
+	}
```
Comment (royiro10) [inline common/logger.go]:
```suggestion
	level := slog.LevelInfo
	if len(levels) > 0 {
		level = levels[0]
	}
```

assigning default first can be more readable and make for compact and smaller code

**Pull request merged**